lib/Basic/LangOptions.cpp

```diff
@@ -389,7 +389,14 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     addPlatformConditionValue(PlatformConditionKind::Arch, "riscv64");
     break;
   default:
-    UnsupportedArch = true;
```

Comment:
Style Nit: I think that if we leave this, it can be simpler - the special case can set this to false instead.  That would avoid some of the indentation.